sys/fuchsia/profile.txt

```diff
@@ -0,0 +1,17 @@
+# Copyright 2022 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+# See https://cs.opensource.google/fuchsia/fuchsia/+/main:zircon/vdso/profile.fidl
+
+include <zircon/syscalls.h>
+
+# Never used as input
```

Comment:
But it is used as output. The FIDL is:

```
    create(resource struct {
        root_job handle:JOB;
        options uint32;
        profile ProfileInfo;
    }) -> (resource struct {
        status status;
        out handle:PROFILE;
    });
```

(3 input arguments, 2 return values)

and the matching C is:

```
    zx_status_t zx_profile_create(zx_handle_t root_job,
                                  uint32_t options,
                                  const zx_profile_info_t* profile,
                                  zx_handle_t* out);
```

(4 arguments, the 4th being the 2nd return value)

So the `zx_profile_create` definition at the bottom of this file should be:

```
zx_profile_create(root_job zx_job, options int32, profile ptr[in, zx_profile_info], out_handle ptr[out, zx_profile])
```

(i.e. change `zx_handle` to `zx_profile`)

Reply:
When we get to the beautiful future where all of {C, C++, Rust, syzlang} system call definitions are automatically generated from FIDL, we won't have to deal with this weirdness anymore. :)

Reply:
Yes, it's definitely used as output.
However when compiling, it's failing:

```
bin/syz-sysgen
compilation of fuchsia/amd64 target failed:
sys/fuchsia/profile.txt:9:1: resource zx_profile is never used as an input (such resources are not useful)
make[1]: *** [Makefile:149: .descriptions] Error 1
```

Reply:
Similar to https://github.com/google/syzkaller/blob/master/sys/fuchsia/pc.txt#L8